instrumentation/github.com/uber-go/zap/otelzap/otelzap.go

```diff
@@ -0,0 +1,153 @@
+func WarnWithSpan(span oteltrace.Span, log string, fields ...zapcore.Field) {
+	Warn(log, fields...)
+	logSpan(span, log, fields...)
+
+}
+
+// Warn logs on warn level.
+func Warn(log string, fields ...zapcore.Field) {
+	zap.L().Warn(log, fields...)
+}
+
+// ErrorWithContext logs on error level and trace based on the context span if it exists.
+func ErrorWithContext(ctx context.Context, log string, fields ...zapcore.Field) {
+	ErrorWithSpan(oteltrace.SpanFromContext(ctx), log, fields...)
+}
+
+// ErrorWithSpan logs on error level and add the logs on the trace if span exists.
+func ErrorWithSpan(span oteltrace.Span, log string, fields ...zapcore.Field) {
+	Error(log, fields...)
+	logSpan(span, log, fields...)
+}
+
+// Error logs on error level.
+func Error(log string, fields ...zapcore.Field) {
+	zap.L().Error(log, fields...)
+}
+
+
+// DPanicWithContext logs on dPanic level and trace based on the context span if it exists.
+func DPanicWithContext(ctx context.Context, log string, fields ...zapcore.Field) {
+	DPanicWithSpan(oteltrace.SpanFromContext(ctx), log, fields...)
+}
+
+// DPanicWithSpan logs on dPanic level and add the logs on the trace if span exists.
+func DPanicWithSpan(span oteltrace.Span, log string, fields ...zapcore.Field) {
+	logSpan(span, log, fields...)
+	DPanic(log, fields...)
+}
+
+// DPanic logs on dPanic level.
+func DPanic(log string, fields ...zapcore.Field) {
+	zap.L().DPanic(log, fields...)
+}
+
+// PanicWithContext logs on panic level and trace based on the context span if it exists.
+func PanicWithContext(ctx context.Context, log string, fields ...zapcore.Field) {
+	PanicWithSpan(oteltrace.SpanFromContext(ctx), log, fields...)
+}
+
+// PanicWithSpan logs on panic level and add the logs on the trace if span exists.
+func PanicWithSpan(span oteltrace.Span, log string, fields ...zapcore.Field) {
+	logSpan(span, log, fields...)
+	Panic(log, fields...)
+}
+
+// Panic logs on panic level.
+func Panic(log string, fields ...zapcore.Field) {
+	zap.L().Panic(log, fields...)
+}
+
+// FatalWithContext logs on fatal level and trace based on the context span if it exists.
+func FatalWithContext(ctx context.Context, log string, fields ...zapcore.Field) {
+	FatalWithSpan(oteltrace.SpanFromContext(ctx), log, fields...)
+}
+
+// FatalWithSpan logs on fatal level and add the logs on the trace if span exists.
+func FatalWithSpan(span oteltrace.Span, log string, fields ...zapcore.Field) {
+	logSpan(span, log, fields...)
+	Fatal(log, fields...)
+}
+
+// Fatal logs on fatal level.
+func Fatal(log string, fields ...zapcore.Field) {
+	zap.L().Fatal(log, fields...)
+}
+
+func logSpan(span oteltrace.Span, log string, fields ...zapcore.Field) {
+	if span != nil {
+		attrs := make([]otellabel.KeyValue, len(fields)+1)
+		if log != "" {
+			attrs = append(attrs, otellabel.String("event", log))
+		}
+		if len(fields) > 0 {
+			attrs = append(attrs, zapFieldsToOtel(fields...)...)
+		}
+		span.SetAttributes(attrs...)
```

Comment:
This should probably use [`span.AddEvent()`](https://github.com/open-telemetry/opentelemetry-go/blob/v0.15.0/trace/trace.go#L256) rather than `SetAtributes`.